gateway-server/src/main/java/org/apache/knox/gateway/websockets/GatewayWebsocketHandler.java

```diff
@@ -117,7 +117,10 @@ public Object createWebSocket(ServletUpgradeRequest req,
       final String backendURL = getMatchedBackendURL(path);
 
       /* Upgrade happens here */
-      return new ProxyWebSocketAdapter(URI.create(backendURL), pool, getClientEndpointConfig(req));
+      ProxyWebSocketAdapter wsAdaptor =  new ProxyWebSocketAdapter
+              (URI.create(backendURL), pool, getClientEndpointConfig(req));
+      wsAdaptor.setMaxBufferCount(config.getWebsocketMaxWaitBufferCount());
```

Comment:
I believe we do not need this API at all; you should just use `config.getWebsocketMaxWaitBufferCount()` within `ProxyWebSocketAdapter` when checking the limit

Reply:
How can config object (GatewayConfigImpl class) be accessed in ProxyWebSocketAdapter ?

Reply:
You can pass it as a constructor argument when creating the `wsAdaptor` instance.

Reply:
Yeah passing config as constructor argument makes more sense. Fixed now in new commit.